mongo/options/clientoptions.go

```diff
@@ -206,6 +226,7 @@ type ClientOptions struct {
 	RetryReads               *bool
 	RetryWrites              *bool
 	ServerAPIOptions         *ServerAPIOptions
+	ServerMonitoringMode     *string
```

Comment:
We also define expected values for the read/write concerns. Not using a pointer here seems to deviate from the existing pattern, which seems to intend that data in an options struct be nullable. We don't expect users to construct this data at the struct level, per  the comment for "ClientOptions" states that 

> Each option can be set through setter functions.

Would you mind expanding on this? And what is meant by "we should define a string value for 'not set'"? If we do go the non-pointer route, I don't think we should extend the constants to include a "ServerMonitoringModeNotSet" case, it's not defined in the [specifications](https://github.com/mongodb/specifications/pull/1452/files#diff-67f5deaee17cfd34b1845819d83009a9b7e16d2ef28ffce946f51b714474efde) and I'm not sure how it would help a user.

Reply:
I was mostly reacting to the additional logic for handling pointers, which is unnecessary if the zero value can represent "not set". However, you're correct that the strong existing pattern in `ClientOptions` is to use pointers and `nil` for "not set". This PR doesn't seem like the right place to change that pattern. Consider my original comment resolved.